src/Native/System.Security.Cryptography.Native/pal_ssl.h

```diff
@@ -6,6 +6,154 @@
 #include <openssl/ssl.h>
 
 /*
+These values should be kept in sync with System.Security.Authentication.CipherAlgorithmType.
+*/
+enum class CipherAlgorithmType : int32_t
+{
+    None = 0,
+    Null = 24576,
+    Des = 26113,
+    Rc2 = 26114,
+    TripleDes = 26115,
+    Aes128 = 26126,
+    Aes192 = 26127,
+    Aes256 = 26128,
+    Aes = 26129,
+    Rc4 = 26625,
+
+    //Algorithm constants which are not present in the managed CipherAlgorithmType enum.
+    SSL_IDEA = 229380,
+    SSL_CAMELLIA128 = 229381,
+    SSL_CAMELLIA256 = 229382,
+    SSL_eGOST2814789CNT = 229383,
+    SSL_SEED = 229384,
+};
+
+/*
+The values used in OpenSSL for SSL_CIPHER algorithm_enc.
+*/
+enum class SSL_CipherAlgorithm : int64_t
+{
+#ifdef HAVE_SSL_CIPHER_SPLIT_ALGORITHMS
+    SSL_DES = 1,
+    SSL_3DES = 2,
+    SSL_RC4 = 4,
+    SSL_RC2 = 8,
+    SSL_IDEA = 16,
+    SSL_eNULL = 32,
+    SSL_AES128 = 64,
+    SSL_AES256 = 128,
+    SSL_CAMELLIA128 = 256,
+    SSL_CAMELLIA256 = 512,
+    SSL_eGOST2814789CNT = 1024,
+    SSL_SEED = 2048,
+    SSL_AES128GCM = 4096,
+    SSL_AES256GCM = 8192
+#else
+    SSL_DES = 0x00008000L,
+    SSL_3DES = 0x00010000L,
+    SSL_RC4 = 0x00020000L,
+    SSL_RC2 = 0x00040000L,
+    SSL_IDEA = 0x00080000L,
+    //SSL_eFZA = 0x00100000L,  this value is defined in ssl_locl.h, but has no match
+    SSL_eNULL = 0x00200000L,
+    SSL_AES = 0x04000000L,
+    SSL_CAMELLIA = 0x08000000L,
+    SSL_SEED = 0x10000000L,
+#endif
+};
+
+/*
+These values should be kept in sync with System.Security.Authentication.ExchangeAlgorithmType.
+*/
+enum class ExchangeAlgorithmType : int32_t
+{
+    None,
+    RsaSign = 9216,
+    RsaKeyX = 41984,
+    DiffieHellman = 43522,
+
+    //ExchangeAlgorithm constants which are not present in the managed ExchangeAlgorithmType enum.
```

Comment:
This is an existing quirk in the code.  The public .NET enum has the first 4 values. But there are more values that OpenSSL can return than what is defined in the current public .NET enum. The existing managed code was already taking these int values and casting them to the enum.  See https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/Unix/libssl/SslConnectionInfo.cs#L138-L157

While I don't think this is a great approach, I didn't want to change the behavior with the shim work. I just ported the behavior to native.

@vijaykota - can you elaborate on the long term thoughts around these enums?